handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -494,6 +494,12 @@ public void flush(ChannelHandlerContext ctx) throws Exception {
         }
         try {
             wrap(ctx, false);
+        } catch (Exception e) {
+            // Just fail the pending writes in case of an SSLException. If there are no pending writes rethrow.
+            if (pendingUnencryptedWrites.removeAndFailAll(e)) {
```

Comment:
Because if we do we will see an exception which is not correct IMHO as we already handled it correctly:

https://github.com/netty/netty/blob/4.1/transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java#L788

Also if we do so we will not call `safeClose(...)` which I think we still should if we was able notify pending writes:
https://github.com/netty/netty/pull/5685/files#diff-2fe7b22a8d650f1ea0bf56a809c061f9R1265